### PR TITLE
feat(textfield): Convert some foundation methods from private to public

### PR DESCRIPTION
**Pull request by bwobrien**
This is to allow integration into other web frameworks.

### CODE REVIEW

File: packages/mdc-textfield/foundation.js [hunk 1]
```diff
@@ -202,9 +198,8 @@ class MDCTextfieldFoundation extends MDCFoundation {
    * Fires when animation transition ends, performing actions that must wait
    * for animations to finish.
    * @param {!Event} evt
-   * @private
    */
-  transitionEnd_(evt) {
+  transitionEnd(evt) {
```
Comment (kfranqueiro) [inline packages/mdc-textfield/foundation.js]: I'm kind of surprised that we'd be publicizing this method in particular... would there be a better name for this that would give better context to when it'd make sense to directly invoke it?

Comment (bwobrien) [reply packages/mdc-textfield/foundation.js]: Gave this and setBottomLineTransformOrigin more semantic names.

File: packages/mdc-textfield/README.md [hunk 2]
```diff
@@ -426,6 +426,32 @@ Returns a boolean specifying whether or not the input is disabled.
 
 Updates the input's disabled state.
 
+##### MDCTextfieldFoundation.setValid(isValid: boolean)
+
+Sets the validity state of the Text Field. Triggers custom validity checking.
+
+##### MDCTextfieldFoundation.handleTextFieldInteraction(evt: !Event)
```
Comment (kfranqueiro) [inline packages/mdc-textfield/README.md]: Nit: FWIW, I'm pretty sure we follow typescript conventions when documenting APIs in READMEs, so the `!`s here might not make sense...

Comment (bwobrien) [reply packages/mdc-textfield/README.md]: Done.

**Pull request merged**